polars/polars-core/src/chunked_array/ops/cum_agg.rs

```diff
@@ -135,6 +135,97 @@ where
         ca.rename(self.name());
         ca
     }
+
+    fn cumcount(&self, reverse: bool, count_nulls: bool) -> UInt32Chunked {
+        let mut initial_value = 0u32;
+        let mut chunked_array: UInt32Chunked = match reverse {
+            false => self
+                .into_iter()
+                .map(|x| {
+                    if x.is_some() || count_nulls {
+                        initial_value += 1;
+                    }
+                    Some(initial_value)
+                })
+                .collect_trusted(),
+            true => self
+                .into_iter()
+                .rev()
+                .map(|x| {
+                    if x.is_some() || count_nulls {
+                        initial_value += 1;
+                    }
+                    Some(initial_value)
+                })
+                .collect_reversed(),
+        };
+        chunked_array.rename(self.name());
+        chunked_array
+    }
+}
```

Comment:
I'd love to reduce some of the duplicated code (this function is implemented three times) but I can't get the trait bounds right

Reply:
I will resolve this. If it is deemed a blocker we can re-open it.